srtcore/core.cpp

```diff
@@ -10360,6 +10344,33 @@ int srt::CUDT::processData(CUnit* in_unit)
             }
         }
 
+        if (!srt_loss_seqs.empty())
+        {
+            ScopedLock lock(m_RcvLossLock);
+
+            HLOGC(qrlog.Debug,
+                  log << CONID() << "processData: LOSS DETECTED, %: " << Printable(srt_loss_seqs) << " - RECORDING.");
+            for (loss_seqs_t::iterator i = srt_loss_seqs.begin(); i != srt_loss_seqs.end(); ++i)
+            {
+                // If loss found, insert them to the receiver loss list.
+                m_pRcvLossList->insert(i->first, i->second);
+            }
+
+            if (initial_loss_ttl)
+            {
+                // pack loss list for (possibly belated) NAK
+                // The LOSSREPORT will be sent in a while.
+                for (loss_seqs_t::iterator i = srt_loss_seqs.begin(); i != srt_loss_seqs.end(); ++i)
+                {
+                    m_FreshLoss.push_back(CRcvFreshLoss(i->first, i->second, initial_loss_ttl));
+                }
+                HLOGC(qrlog.Debug,
+                      log << CONID() << "FreshLoss: added sequences: " << Printable(srt_loss_seqs)
+                          << " tolerance: " << initial_loss_ttl);
+                reorder_prevent_lossreport = true;
+            }
+        }
```

Comment:
`reorder_prevent_lossreport` basically has the same value as `initial_loss_ttl`. Therefore we don't need `reorder_prevent_lossreport `.
- [ ] Unless I am missing something, please replace `reorder_prevent_lossreport ` with `initial_loss_ttl`.

It might be worth reusing the `for` loop, as it may be more CPU-expensive than the extra check inside.

```suggestion
        if (!srt_loss_seqs.empty())
        {
            ScopedLock lock(m_RcvLossLock);

            HLOGC(qrlog.Debug,
                  log << CONID() << "processData: LOSS DETECTED, %: " << Printable(srt_loss_seqs) << " - RECORDING.");
            for (loss_seqs_t::iterator i = srt_loss_seqs.begin(); i != srt_loss_seqs.end(); ++i)
            {
                m_pRcvLossList->insert(i->first, i->second);
                if (initial_loss_ttl)
                {
                    m_FreshLoss.push_back(CRcvFreshLoss(i->first, i->second, initial_loss_ttl));
                }
            }
        }
```